packages/core/src/actions/execute.ts

```diff
@@ -417,25 +415,95 @@ export const executeActionsViaManagedService: ExecuteActions = async (
     signer
   )
 
+  const sphinxModule = new ethers.Contract(
+    moduleAddress,
+    SphinxModuleABI,
+    provider
+  )
+
+  const executionData = sphinxModule.interface.encodeFunctionData('execute', [
+    batch,
+  ])
+  const managedServiceExecData = managedService.interface.encodeFunctionData(
+    'exec',
+    [moduleAddress, executionData]
+  )
+
+  const overrides: ethers.TransactionRequest = {}
+  if (
+    shouldBufferExecuteActionsGasLimit((await provider.getNetwork()).chainId)
+  ) {
+    const minimumActionGas = estimateGasViaManagedService(moduleAddress, batch)
+    const gasEstimate = await signer.estimateGas({
+      to: getManagedServiceAddress(),
+      data: managedServiceExecData,
+    })
+
+    const block = await provider.getBlock('latest')
```

Comment:
> This is optional, but both executeActionsViaSigner and executeActionsViaManagedService call await provider.getNetwork in every call, which is inefficient. I think we already have the chainId available as a variable in the functions that call into these executeActions functions, so it'd be easy to pass it in as a parameter, which'd save 1 RPC call per batch of execution actions

Likewise for `provider.getBlock`. (We already have the `blockGasLimit` on the functions that call these execution functions)